src/solvers/hard_coded_ars343.jl

```diff
@@ -0,0 +1,165 @@
+function step_u!(integrator, cache::IMEXARKCache, ::ARS343)
+    (; u, p, t, dt, sol, alg) = integrator
+    (; f) = sol.prob
+    (; T_imp!, lim!, dss!) = f
+    (; tableau, newtons_method) = alg
+    (; a_exp, b_exp, a_imp, b_imp, c_exp, c_imp) = tableau
+    (; U, T_lim, T_exp, T_imp, temp, γ, newtons_method_cache) = cache
+    T_lim! = !isnothing(f.T_lim!) ? f.T_lim! : (args...) -> nothing
+    T_exp! = !isnothing(f.T_exp!) ? f.T_exp! : (args...) -> nothing
+
+    if !isnothing(newtons_method_cache)
+        jacobian = newtons_method_cache.j
+        if (!isnothing(jacobian)) &&
+           needs_update!(newtons_method.update_j, newtons_method_cache.update_j_cache, NewTimeStep(t))
+            T_imp!.Wfact(jacobian, u, p, dt * γ, t)
+        end
+    end
+
+    implicit_equation_residual_general! =
+        (residual, Ui, ii) -> begin
+            T_imp!(residual, Ui, p, t + dt * c_imp[ii]) #= t_imp = =#
+            @. residual = temp + dt * a_imp[ii, ii] * residual - Ui
+        end
+    implicit_equation_jacobian_general! =
+        (jacobian, Ui, ii) -> T_imp!.Wfact(jacobian, Ui, p, dt * a_imp[ii, ii], t + dt * c_imp[ii]) #= t_imp = =#
+
+    s = 4
+
+    i::Int = 1
+    t_exp = t
+    @. U[i] = u
+    lim!(U[i], p, t_exp, u)
+    dss!(U[i], p, t_exp)
+    T_lim!(T_lim[i], U[i], p, t_exp)
+    T_exp!(T_exp[i], U[i], p, t_exp)
+
+    i = 2
+    t_exp = t + dt * c_exp[i]
+    @. U[i] = u + dt * a_exp[i, 1] * T_lim[1]
+    lim!(U[i], p, t_exp, u)
+    @. U[i] += dt * a_exp[i, 1] * T_exp[1]
+    dss!(U[i], p, t_exp)
+    @. temp = U[i] # used in closures
+    let i = i
+        implicit_equation_residual! = (residual, Ui) -> implicit_equation_residual_general!(residual, Ui, i)
+        implicit_equation_jacobian! = (jacobian, Ui) -> implicit_equation_jacobian_general!(jacobian, Ui, i)
+        solve_newton!(
+            newtons_method,
+            newtons_method_cache,
+            U[i],
+            implicit_equation_residual!,
+            implicit_equation_jacobian!,
+        )
+    end
+
+    @. T_imp[i] = (U[i] - temp) / (dt * a_imp[i, i])
+
+    T_lim!(T_lim[i], U[i], p, t_exp)
+    T_exp!(T_exp[i], U[i], p, t_exp)
+
+    i = 3
+    t_exp = t + dt * c_exp[i]
+    @. U[i] = u + dt * a_exp[i, 1] * T_lim[1] + dt * a_exp[i, 2] * T_lim[2]
+    lim!(U[i], p, t_exp, u)
+    @. U[i] += dt * a_exp[i, 1] * T_exp[1] + dt * a_exp[i, 2] * T_exp[2] + dt * a_imp[i, 2] * T_imp[2]
+    dss!(U[i], p, t_exp)
+
+    @. temp = U[i] # used in closures
+    let i = i
+        implicit_equation_residual! = (residual, Ui) -> implicit_equation_residual_general!(residual, Ui, i)
```

Comment:
I think it would be easier to just define the functions here. I realize its duplicated, but it makes it clearer what is going on